src/Repository/CreditRepository.php

```diff
@@ -142,8 +143,31 @@ public function getImageCredits( array $images ) : array {
 					)
 				  ) a"
 			);
-		$connection->close();
+	}
+
+	/**
+	 * Runs Connection::fetchAllAssociative() on the given SQL and closes the Connection.
+	 * Also captures common MySQL errors where we want to tell the user to re-try.
+	 * @param Connection $connection
+	 * @param string $sql
+	 * @return array
+	 * @throws WsExportException
+	 * @throws \Doctrine\DBAL\Exception
```

Comment:
I think we're only adding directly-thrown exceptions with `@throws` (i.e. WsExportException in this case, but not this other one), or at least that's what I've been doing!